Embedded market maker struct in smt_token_object

Kiwonik opened the pull request:
The new member replaced `current_supply` member too. #2029

youkaicountry commented:
There seems to be a miscommunication, the `token_balance` member represents the balance of the token held by the market (t in the paper), not the total supply. The supply should still be held in the `current_supply` member of the object. 

I removed the member from the original spec that represented total supply (T in the paper) from this struct.

Kiwonik commented:
Yes, that exactly what happened, i removed a member that has already been removed from (now) embedded structure. A commit restoring the member I removed has already passed the tests. Ready to merge again.
